mutiny-core/src/lib.rs

```diff
@@ -2189,6 +2197,46 @@ impl<S: MutinyStorage> MutinyWallet<S> {
     pub fn is_safe_mode(&self) -> bool {
         self.safe_mode
     }
+
+    /// Calls upon a Cashu mint and withdrawls/redeem/melts the token from it.
+    pub async fn melt_cashu_token(&self, token_v3: TokenV3) -> Result<bool, MutinyError> {
+        let mint_client = CrossPlatformHttpClient::new();
+        let mut paid = false;
+
+        for token in token_v3.tokens {
+            let mint_url = match token.mint {
+                Some(url) => url,
+                None => return Err(MutinyError::EmptyMintURLError),
+            };
+
+            let invoice_amount = token.proofs.total_amount() * (1 - 3 / 100);
```

Comment:
Im not sure that `(1 - 3 / 100)` will actually do anything, this might just make end up as `1` because they are all `u64`s. Might need to do it like `let invoice_amount = token.proofs.total_amount() as f64 * (1.0 - 3.0 / 100.0);`